### PR TITLE
Refactoring datasource v2 to use the BQ storage API v1

### PR DESCRIPTION
**Pull request by davidrabinowitz**

### CODE REVIEW

File: connector/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryClient.java [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+
+    BigQueryClient(BigQuery bigQuery, Optional<String> materializationProject, Optional<String> materializationDataset) {
+        this.bigQuery = bigQuery;
+        this.materializationProject = materializationProject;
+        this.materializationDataset = materializationDataset;
+    }
+
+    // return empty if no filters are used
+    private static Optional<String> createWhereClause(String[] filters) {
+        return Optional.empty();
```
Comment (Gaurangi94) [inline connector/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryClient.java:57]: shouldn't we be returning filters here?

File: connector/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryClient.java [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+    }
+
+    public TableInfo getTable(TableId tableId) {
+        TableId bigQueryTableId = tableIds.get(tableId);
```
Comment (Gaurangi94) [inline connector/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryClient.java:61]: how are these two tableIds related?

File: connector/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryClient.java [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+        DatasetId bigQueryDatasetId = datasetIds.getOrDefault(datasetId, datasetId);
+        Iterable<Table> allTables = bigQuery.listTables(bigQueryDatasetId).iterateAll();
+        return StreamSupport.stream(allTables.spliterator(), false)
+                .filter(table -> allowedTypes.contains(table.getDefinition().getType()))
+                .collect(toImmutableList());
+    }
+
+    private Dataset addDataSetMappingIfNeeded(Dataset dataset) {
+        DatasetId bigQueryDatasetId = dataset.getDatasetId();
+        DatasetId prestoDatasetId = DatasetId.of(bigQueryDatasetId.getProject(), bigQueryDatasetId.getDataset().toLowerCase(ENGLISH));
```
Comment (Gaurangi94) [inline connector/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryClient.java:119]: I don't understand the relation between presto and bigquery datasetIds here

File: connector/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryErrorCode.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    // Should be last
+    UNSUPPORTED(9998),
+    UNKNOWN(9999);
+
+    final int code;
+
+    BigQueryErrorCode(int code) {
+        this.code = code;
+    }
+
```
Comment (Gaurangi94) [inline connector/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryErrorCode.java:31]: nit : extra new line

File: connector/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryReadClientFactory.java [hunk 5]
```diff
@@ -0,0 +1,73 @@
+/*
```
Comment (Gaurangi94) [inline connector/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryReadClientFactory.java:1]: not: two copyright statements here

File: connector/src/main/java/com/google/cloud/spark/bigquery/BigQueryDataSourceReader.java [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+    @Override
+    public void pruneColumns(StructType requiredSchema) {
+        this.schema = Optional.ofNullable(requiredSchema);
+    }
+
+    Optional<String> emptyIfNeeded(String value) {
+        return (value == null || value.length() == 0) ?
+                Optional.empty() : Optional.of(value);
+    }
+
```
Comment (Gaurangi94) [inline connector/src/main/java/com/google/cloud/spark/bigquery/BigQueryDataSourceReader.java:178]: nit: extra new line

File: connector/src/main/java/com/google/cloud/spark/bigquery/BigQueryEmptyProjectionInputPartitionReader.java [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+
+    @Override
+    public InternalRow get() {
+        currentIndex++;
+        return InternalRow.empty();
+    }
+
+    @Override
+    public void close() throws IOException {
+        // empty
```
Comment (Gaurangi94) [inline connector/src/main/java/com/google/cloud/spark/bigquery/BigQueryEmptyProjectionInputPartitionReader.java:46]: why is this empty?

File: connector/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryClient.java (hunk 1 above)
Comment (davidrabinowitz) [reply connector/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryClient.java:57]: Thanks, I've fixed the bug

File: connector/src/main/java/com/google/cloud/spark/bigquery/BigQueryEmptyProjectionInputPartitionReader.java (hunk 7 above)
Comment (davidrabinowitz) [reply connector/src/main/java/com/google/cloud/spark/bigquery/BigQueryEmptyProjectionInputPartitionReader.java:46]: This class is an iterator that returns empty rows based on the size. As it does not rely on any external resource there is nothing to close.

File: connector/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryClient.java (hunk 3 above)
Comment (davidrabinowitz) [reply connector/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryClient.java:119]: Should have been removed
